### PR TITLE
Add example for lazySource

### PR DESCRIPTION
**Pull request by mggger**
References #25468

### CODE REVIEW

**Comment by mggger** [comment]
@hepin1989 hi,  it was my first time to submit a pr, please help me to know what's left to do

**Comment by chbatey** [comment]
OK TO TEST

File: akka-docs/src/test/scala/docs/stream/operators/source/LazySource.scala [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  implicit val system: ActorSystem = ???
+
+  def fib(n: Int): Long = n match {
+    case x if x < 0 =>
+      throw new IllegalArgumentException("Only positive numbers allowed")
+    case 0 | 1 => 1
+    case _     => fib(n - 2) + fib(n - 1)
+  }
+
+  val lzSource: Source[() => Long, NotUsed] = Source((0 to 10).map(i => {
```
Comment (chbatey) [inline akka-docs/src/test/scala/docs/stream/operators/source/LazySource.scala:19]: This should use Source.lazySource for the example

**Comment by johanandren** [comment]
Hi @mggger thanks for giving this one a shot, I'm afraid the lazy operators are somewhat tricky in where they are useful and what is a good example, so I picked them up just now. 

If you still are interested I'd recommend picking operators with more clear use case, for example `.alsoTo`, `.asInputStream`, `.asOutputStream`.

**Comment by mggger** [comment]
@johanandren  Ok,  I will have a try about the operators you recommend

**Pull request closed without merging**